go/adbc/driver/snowflake/record_reader.go

```diff
@@ -266,6 +282,24 @@ func getTransformer(sc *arrow.Schema, ld gosnowflake.ArrowStreamLoader, useHighP
 	return out, getRecTransformer(out, transformers)
 }
 
+func integerToDecimal128(ctx context.Context, a arrow.Array, dt *arrow.Decimal128Type) (arrow.Array, error) {
+	// We can't do a cast directly into the destination type because the numbers we get from Snowflake
+	// are scaled integers. So not only would the cast produce the wrong value, it also risks producing
+	// an error of precisions which e.g. can't hold every int64. To work around these problems, we instead
+	// cast into a decimal type of a precision and scale which we know will hold all values and won't
+	// require scaling, We then substitute the type on this array with the actual return type.
```

Comment:
I hate this so much, but it makes sense that we have to do this because of how snowflake works.